src/modules/universal-discovery/components/finder/finder.component.js

```diff
@@ -18,39 +19,142 @@ export default class FinderComponent extends Component {
+                Object.assign({}, this.props.restInfo, { parentLocationId: this.props.startingLocationId }),
+                this.updateLocationsData
+            );
+        }
     }
 
     componentDidUpdate() {
         this.updateBranchesContainerScroll();
     }
 
+    componentWillReceiveProps(props) {
+        if (this.preselectedItem && this.locationsMap && this.activeLocations && props.isVisible && props.isVisible !== this.props.isVisible) {
+            this.setState(state => Object.assign({}, state, {
+                locationsMap: this.locationsMap,
+                activeLocations: this.activeLocations
+            }), () => this.props.isVisible && this.props.onItemSelect(this.preselectedItem));
+        }
+    }
+
+    /**
+     * Load data of preselected location.
+     *
+     * @method loadPreselectedData
+     * @param {String} locationId
+     * @memberof FinderComponent
+     */
+    loadPreselectedData(id) {
+        const promise = new Promise(resolve => loadAccordionData(
+            {startingLocationId: this.props.startingLocationId, locationId: id},
+            resolve
+        ));
+
+        promise
+            .then((response) => {
+                this.createAccordionData(response);
+                this.preselectedItem = response.locations[id].Location;
+                this.setAccordionData();
+            });
+    }
+
+    /**
+     * Create an accordion data.
+     *
+     * @method createAccordionData
+     * @param {Object} locations
+     * @param {Object} subitems
+     * @memberof FinderComponent
+     */
+    createAccordionData({locations, subitems}) {
+        const createItem = ({parent, offset = 0, children, location}) => {
+            const createData = (items) => {
+                return items.locations.map(location => {
+                    return {value: location};
+                });
+            };
+            const data = children ? createData(children) : [];
+            const count = children ? children.totalCount : location.childCount;
+
+            return {count, data, offset, parent};
+        };
+
+        if (subitems[1]) {
+            const item = createItem({
+                parent: 1,
+                children: subitems[1]
+            });
+
+            this.locationsMap[1] = item;
+            this.activeLocations[0] = item;
+        }
+
+        Object.entries(locations).forEach(([key, value]) => {
+            const item = createItem({
+                parent: parseInt(key, 10),
+                children: subitems[key],
+                location: value.Location
+            });
+
+            this.locationsMap[key] = item;
+            this.activeLocations[value.Location.depth] = item;
+        });
+    }
+
+    /**
+     * Sets an accordion data.
+     *
+     * @method setAccordionData
+     * @memberof FinderComponent
+     */
+    setAccordionData() {
+        this.setState(state => Object.assign({}, state, {
+            locationsMap: this.locationsMap,
```

Comment:
Do you want to pass the references to properties this way? Maybe you should make these data immutable to avoid unexpected behaviours in future.